### ISSUE TITLE
[Bug] Inconsistent result when using an Image Gallery from a Shared Subscription

### ISSUE THREAD

**Issue by j7lloyd** [citrix/terraform-provider-citrix#117]
## Describe the bug

The issue arises when using a `gallery_image` block in combination with the `shared_subscription` attribute while defining an `azure_master_image`. From my understanding, the expected behavior is impaired due to the way the `MasterImageVM` property is broken down to populate the required attributes. Specifically, the problem appears to stem from the introduction of an extra element in the path: `\<subscription_id>.sharedsubscription`.

**Without** a shared subscription, the path is formatted as follows:

> XDHyp:\HostingUnits\RegionAndNetworkResources\image.folder\ResourceGroup.resourcegroup\GalleryName.gallery\ImageDefinition.imagedefinition\1.0.0.imageversion

However, **with** a shared subscription, the path becomes:

> XDHyp:\HostingUnits\RegionAndNetworkResources\image.folder\SubscriptionID.sharedsubscription\ResourceGroup.resourcegroup\GalleryName.gallery\ImageDefinition.imagedefinition\1.0.0.imageversion

The additional element (`\SubscriptionID.sharedsubscription`) causes issues in interpreting and parsing the path correctly.

## Versions

**Terraform:** v1.9.5
**citrix/citrix provider:** v1.0.1
**Operation system:** linux_amd64

## Terraform configuration files

Below is the relevant snippet from the affected `citrix_machine_catalog` resource:

```
azure_master_image = {
  shared_subscription = "<subscription_id>"
  resource_group      = "<resource_group_name>"
  gallery_image = {
    gallery    = "<compute_gallery_name>"
    definition = "<image_definition_name>"
    version    = "<image_version_name>"
  }
}
```

## Terraform console output

│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to citrix_machine_catalog.main, provider "provider[\"registry.terraform.io/citrix/citrix\"]" produced an unexpected new value: .provisioning_scheme: inconsistent values 
│ for sensitive attribute.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.

**Comment by j7lloyd** [comment]
As suspected, the current implementation does not handle segment lengths (`lastIndex`) greater than 8 when refreshing properties for the `masterImage`. Consequently, the following block of code needs to be extended to support the additional `.sharedsubscription` element, which introduces a 9th segment and triggers processing by the `else` statement: https://github.com/citrix/terraform-provider-citrix/blob/50cace8b72c49bce1c1d3bada6ab437ad22688d2/internal/daas/machine_catalog/machine_config.go#L1089-L1121

**Comment by aneeshk-citrix** [comment]
Hi @j7lloyd,

Thank you bringing up the issue and also pointing it out in the implementation. We will work on a fix make this available as soon as we can.

Thanks,
Aneesh

**Comment by TimBennett13** [comment]
Hello,
I am still getting the error following a Machine Catalog creation. The machine catalog is created and can be destroyed.
 Error: Provider produced inconsistent result after apply
│
│ When applying changes to citrix_machine_catalog.DaaSMachineCatalog, provider "provider[\"registry.terraform.io/citrix/citrix\"]" produced an unexpected new value: .provisioning_scheme: inconsistent values for      
│ sensitive attribute.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.

Terraform v1.7.5
on windows_amd64
+ provider registry.terraform.io/citrix/citrix v1.0.3
+ provider registry.terraform.io/hashicorp/azurerm v4.2.0

**Comment by AlanCitrix** [comment]
Ok, reviewing your logs in #120 closer, I see that `writeback_cache` also may have a bug in it. I apologize for not seeing this originally. Can you try it without the `writeback_cache` and see if it works then?

I opened #124 to track this new issue, will have a fix in the next release for it.

**Comment by j7lloyd** [comment]
@AlanCitrix, just a quick observation. Since the release of 1.0.3, while my original issue with the `shared_subscription` argument has been resolved, strangely, omitting this attribute now produces an inconsistent result. Specifically, as shown in the sample below, it expects an empty string (`""`). That is, omitting the value is treated as setting it to `null`, which is no longer permitted.

```
    ~ provisioning_scheme      = {
          ~ azure_machine_config           = {
              ~ azure_master_image        = {
                  - shared_subscription = "" -> null
```

As alluded to, I can work around this by setting the value to `""`, but could you please consider a more robust fix in the future?

**Comment by zhuolun-citrix** [comment]
Hi @j7lloyd ,

We are aware of this issue and a fix is deployed in the upcoming `v1.0.4` release.

Thank you for reporting this issue!

Zhuolun
